wire-ios/Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/RemoveClients/RemoveClientsViewModel.swift

```diff
@@ -48,25 +45,16 @@ extension RemoveClientsViewController {
                 })
         }
 
-        func removeUserClient(_ userClient: UserClient) async throws {
+        func removeUserClient(_ userClient: UserClient, password: String) async throws {
             let clientId = await userClient.managedObjectContext?.perform {
                 return userClient.remoteIdentifier
             }
             guard let clientId else {
                 throw RemoveUserClientError.clientDoesNotExistLocally
             }
-            try await removeUserClientUseCase?.invoke(clientId: clientId, credentials: credentials?.emailCredentials)
-        }
-    }
-}
 
-private extension ZMEmailCredentials {
-    var emailCredentials: EmailCredentials? {
-        guard let email,
-              let password
-        else {
-            return nil
+            try await removeUserClientUseCase?.invoke(clientId: clientId,
+                                                      credentials: EmailCredentials(email: "", password: password))
```

Comment:
a bit weird that the email is not used, does it make sense to change the interface to accept the password instead?

Reply:
changed